app/src/androidTest/java/com/example/habitsmasher/MainActivityTest.java

```diff
@@ -2084,6 +2085,34 @@ public void forgotPassword_validEmailNotRegistered_emailNotSent() {
         assertTextOnScreen(EMAIL_DOES_NOT_EXIST);
     }
 
+    @Test
+    public void autoCompleteTextView_PartOfValidUsername_usernameSelected() {
+        logInTestUser();
+
+        // Asserts that the current activity is the MainActivity. Otherwise, show “Wrong Activity”
+        _solo.assertCurrentActivity(WRONG_ACTIVITY_MESSAGE, MainActivity.class);
+
+        // click on the Notifications tab in the bottom navigation bar
+        _solo.clickOnView(_solo.getView(R.id.navigation_notifications));
+
+        // ensure that the app has transitioned to the Profile screen
+        assertTextOnScreen(PROFILE_TEXT);
+
+        // click follow user search button
+        _solo.clickOnView(_solo.getView(R.id.follow_user_search_button));
+
+        // click on the auto complete text view
+        _solo.clickOnView(_solo.getView(R.id.auto_complete_text_view));
+
+        // enter part of the users username we wish to follow
+        _solo.enterText(_solo.getEditText("Username"), TEST_USER_SUBSTRING);
+
+        // click on the user name that we wish to follow
+        _solo.clickOnText(TEST_USER_USERNAME);
+
+        assertTextOnScreen(TEST_USER_USERNAME);
```

Comment:
Instead of asserting what text is on screen, can we assert on the value of the text view? After you click on the username from the list, the autocomplete text view should have the value of the username you just clicked.